### DISCUSSION TITLE
Does using index files have an impact on performance, startup time, or memory usage?

### DISCUSSION THREAD

**Discussion by mrshahzeb7** [infinitered/ignite#2498] [Q&A] [answered]
I came across a discussion on LinkedIn where someone suggested that we should avoid using index files as they might increase the build size and initial response time. 
This is why I'm asking the question

I personally never faced any performance issues by using this approach

The only issue I faced was circular dependency because I was using it wrong

Are there any drawbacks to this approach?

**Comment by jamonholmgren** [comment, marked as answer]
There is! https://github.com/infinitered/ignite/blob/master/boilerplate/app/components/index.ts

Where are you seeing it missing?

**Reply by mrshahzeb7** [reply]
I apologize for the oversight. I've been preoccupied with a lot on my mind, which caused me to miss these details. In fact, I added an index file to assets, icons, images, and other places.
And that's why I forgot about it.

I also updated the icon component code, because I prefer to use icons like this instead of passing the name
icon={icons.logo}


Furthermore, I came across a discussion on LinkedIn where someone suggested that we should avoid using index files as they might increase the build size and initial response time. This is why I'm asking the question:

Does using index files have an impact on performance, startup time, or memory usage?
